vcloud/box/common/bootstrap.erb

```diff
@@ -1,26 +1,31 @@
 #!/bin/bash
 
 if [ $1 = "postcustomization" ]; then
-  echo 'ubuntu:<%= vars[:pass_hash] %>' | chpasswd -e 
-  
+  echo 'ubuntu:<%= vars[:pass_hash] %>' | chpasswd -e
+
   dpkg-reconfigure openssh-server
   set -e
   exec >/var/log/userdata.log 2>&1
   export DEBIAN_FRONTEND=noninteractive
   E="/etc/environment"
   echo "LC_ALL=en_GB.UTF-8" > $E
-  
-  R="/etc/resolv.conf"
-  echo "nameserver 8.8.8.8" >$R
-  echo "nameserver 8.8.4.4" >>$R
-  
+
+  <% if vars[:dns_suffix] %>
+    sed -i '/^127.0.1.1/d' /etc/hosts
+    sed -i '/^dns-search/d' /etc/network/interfaces
+    sed -i '/^dns-nameservers/d' /etc/network/interfaces
+    echo -e 'dns-search\t<%= vars[:dns_suffix] %>\ndns-nameservers\t8.8.8.8 8.8.4.4' >> /etc/network/interfaces
+    echo -e 'nameserver 8.8.8.8\nnameserver 8.8.4.4\nsearch <%= vars[:dns_suffix] %>' | resolvconf -a eth0.inet
+  <% end %>
+
```

Comment:
When it didn't exist, the ::fqdn fact wasn't populated correctly. When it did, it was. VMware Tools should have been sorting this out for us even when the fact didn't exist and yet it didn't do.

Whilst I'm happy to try removing this, it adds additional barriers to getting this merged for little benefit. The benefit it realises can be introduced after this is merged, if desired, by way of a separate pull request.